torch/_inductor/kernel/flex_attention.py

```diff
@@ -1542,12 +1649,30 @@ def flex_attention(
                 "num_buffers_warp_spec", num_buffers_warp_spec
             )
 
+        # Disabling TMA by default, only explicit kernel_options supported for now
+        cur_kernel_options.setdefault("USE_TMA", False)
+
         cur_kernel_options.setdefault("BLOCK_M", BLOCK_M)
         cur_kernel_options.setdefault("BLOCK_N", BLOCK_N)
         # Blocksparse options
         cur_kernel_options.setdefault("SPARSE_Q_BLOCK_SIZE", SPARSE_Q_BLOCK_SIZE)
         cur_kernel_options.setdefault("SPARSE_KV_BLOCK_SIZE", SPARSE_KV_BLOCK_SIZE)
 
+        # Collecting input shapes required for calculating num_programs
+        batch_size, seq_len, n_heads, head_dim = query.get_size()
+
+        workspace_arg = None
+        if cur_kernel_options["USE_TMA"]:
+            workspace_arg = get_tma_workspace_arg(
+                num_tma_descriptors=3,
+                device=query.get_device(),
+                num_programs=(
+                    (seq_len + cur_kernel_options["BLOCK_M"] - 1)
```

Comment:
nit can we re-ruse the grid_func from above to make sure they stay in check

Reply:
Updated